### DISCUSSION TITLE
Testing mutations separately with jest

### DISCUSSION THREAD

**Discussion by antonykamp** [blitz-js/blitz#2363] [Q&A] [answered]
Hey,

I was wondering if it's possible to test mutations separately, eg. with jest. This is a problem, when `resolver.authorize()` occurs in `resolver.pipe()`. 

Is it possible to "skip" the authorization without mocking the whole resolver? Could [BlitzProvider](https://blitzjs.com/docs/testing#blitz-provider) help in this case? On the other hand, it shouldn't be possible, because bad boys could do this to access the application too :/ 

I'm looking forward to your answers :)
antony

**Comment by flybayer** [comment, marked as answer]
Yes! New apps already have a couple of these for forgotPassword and resetPassword: https://github.com/blitz-js/blitz/blob/canary/packages/generator/templates/app/app/auth/mutations/resetPassword.test.ts

You will need to add a bit more to `mockCtx` like `$authorize: () => {}` and `userId` or other fields if you need them.

**Reply by antonykamp** [reply]
Thank you! I didn't know that `resolver.authorize` uses `session.$authorize`, but it makes sense :D
